[AArch64] Reduce the costs of and/or/xor reductions

davemgreen opened the pull request:
Since the costs were added the codegen for i8/i16 and/or/xor reductions has improved. This updates the cost model to produce the same costs in terms of number of instructions.

c-rhodes reviewed (approved):
couple of minor comments but otherwise LGTM cheers

davemgreen committed:
[AArch64] Reduce the costs of and/or/xor reductions

Since the costs were added the codegen for i8/i16 and/or/xor reductions has
improved. This updates the cost model to produce the same costs in terms of
number of instructions.
